client/maps/map-thumbnail.tsx

```diff
@@ -261,7 +265,7 @@ export function MapThumbnail({
           onClick={onPreview}
         />
       ) : null}
-      {onToggleFavorite ? (
+      {onToggleFavorite && isLoggedIn ? (
```

Comment:
Wouldn't it make more sense to deal with this at the place passing `onToggleFavorite`? This component is not connected to the store currently and that was an intentional design